### DISCUSSION TITLE
Custom format for unmarshalling doesn't work

### DISCUSSION THREAD

**Discussion by heidiLinguist** [p1c2u/openapi-core#778] [Q&A] [answered]
Hello, Thanks for the openap-core project; it's very helpful. I'm using it to validate my api input. I'm trying to make sure that required values that are blank strings are not allowed. I've tried using the documented YAML specs directly (https://swagger.io/docs/specification/data-models/data-types/#string):

```
type: string
minLength: 1

```
and 
```
  type: string
  pattern: '^.+$'

```
But the unmarshaller ignores these specifications. 

I upgraded to 0.19.0a1 to add a custom validator, per https://openapi-core.readthedocs.io/_/downloads/en/latest/pdf/ by @p1c2u Artur Maciag but his example doesn't seem to do anything:

```
    from datetime import datetime

    def unmarshal_usdate(value):
        return datetime.strptime(value, "%m/%d/%y").date

    extra_format_unmarshallers = {
        "usdate": unmarshal_usdate,
    }

    def validate_usdate(value):
        return bool(re.match(r"^\d{1,2}/\d{1,2}/\d{4}$", value))

    extra_format_validators = {
        "usdate": validate_usdate,
    }

    config = Config(
        extra_format_validators=extra_format_validators,
        extra_format_unmarshallers=extra_format_unmarshallers,
    )

    return OpenAPI.from_file_path(spec_file_path, config=config)

```
The unmarshaller for the YAML:
```
                test:
                    type: string
                    description: Test
                    format: usdate

```
Happily returns:

```
RequestUnmarshalResult(errors=[], body={'test': 'not a date', 'description': 'A multiple choice activity focusing on synonyms', 'binary_result': True, 'directionality': 'passive', 'modality': 'text', 'interaction_type': 'match', 'activity_context': 'synonym', 'activity_visibility': 'public', 'activity_type_name': 'ok ok ok'}, parameters=Parameters(query={}, header={}, cookie={}, path={}), security={})
```
tldr; it would be nice to get this custom validator working, but can anyone give me a workaround in the meantime to make sure a string isn't blank?

Thanks,
Heidi

**Comment by p1c2u** [comment, marked as answer]
Hi @heidiLinguist 

thanks for the report, I prepared fix for that. It's part of 0.19.0a2 version

**Reply by heidiLinguist** [reply]
Thank you!
